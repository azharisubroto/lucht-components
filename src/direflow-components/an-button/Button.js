import React, { useContext } from 'react'
import PropTypes from 'prop-types'
import { EventContext, Styled } from 'direflow-component'
import styles from './Button.css'

const Button = (props) => {
  const dispatch = useContext(EventContext)

  const handleClick = () => {
    const event = new Event('an-button-event')
    dispatch(event)
  }

  return (
    <Styled styles={styles}>
      <button onClick={handleClick} className={`an-button${props.color ? ` ${props.color}` : ''}`}>
        {props.content}
      </button>
    </Styled>
  )
}

Button.defaultProps = {
  color: 'primary',
  content: 'My Button'
}

Button.propTypes = {
  content: PropTypes.any,
  color: PropTypes.string
}

export default Button
