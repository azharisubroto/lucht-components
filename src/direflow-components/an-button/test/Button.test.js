import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import Button from '../Button'

const reactProps = {
  content: 'Button Text',
  color: 'primary'
}

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<Button {...reactProps} />, div)
  ReactDOM.unmountComponentAtNode(div)
})

it('matches snapshot as expected', () => {
  const renderTree = renderer.create(<Button {...reactProps} />)
  expect(renderTree).toMatchSnapshot()
})
