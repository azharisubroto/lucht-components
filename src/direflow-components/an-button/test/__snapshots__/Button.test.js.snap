// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot as expected 1`] = `
<button
  className="an-button primary"
  onClick={[Function]}
>
  Button Text
</button>
`;
